### PR TITLE
[vllm_omni, rollout, trainer, reward, fsdp, algo, tests, doc] feat: add Qwen-Image-Edit-2511 FlowGRPO training support

### PR DESCRIPTION
**Pull request by NancyFyong**
## What does this PR do?

Adds end-to-end FlowGRPO training support for Qwen-Image-Edit-2511 / Qwen-Image-Edit-Plus style image-editing diffusion models in VeRL-Omni. This is a re-submission of #148, previously closed because the original implementation did not converge. The convergence issue has been fixed and validated with a successful training run.

### Main changes

- A Qwen-Image-Edit FlowGRPO pipeline with diffusers training and vLLM-Omni rollout adapters.
- Image-edit-specific prompt, condition-image, latent, log-prob, and timestep handling.
- Qwen-Image-Edit LoRA training and smoke-test entrypoints.
- Image-edit reward utilities, including PickScore reward (local inference mode) and vLLM-backed quality reward support.
- Trainer, rollout, FSDP, config, test, and documentation updates needed by the new path.
- Auto-patch for Hugging Face processor configs that ship without `model_type` (Qwen-Image-Edit variants).

## Checklist Before Starting

- [x] Search for similar PRs. Previous attempts #148 and #147 were closed by maintainer @zhtmike with "open the PR once it is ready". No other open PR addresses Qwen-Image-Edit FlowGRPO training.
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}`: `vllm_omni, rollout, trainer, reward, fsdp, algo, tests, doc`
  - `{type}`: `feat`
  - No breaking API changes.

## Test

> Validated by a real training run on 8×H200 with Qwen-Image-Edit-2511 + LoRA (r=64) + PickScore reward. Reward is monotonically increasing over 140+ steps, confirming the pipeline converges.

### Convergence evidence (addresses #148 review feedback)

| step | reward mean |
|------|------------|
| 16   | 0.775      |
| 100  | 0.81       |
| 138  | 0.83       |

### Training curves

image

### Validation comparison (same Ghibli-style prompt across steps)

| step 20   | reward 0.78 |
image
| step 60   | 0.75 |
image
| step 100  | 0.84 |
image
| step 130  | 0.89 |
image

### Eval curves

image
### Test commands run

```bash
# End-to-end smoke test (tiny model, 1 GPU, 1 step)
NUM_GPUS=1 MODEL_PATH=tiny-random/qwen-image-edit-plus \
bash tests/special_e2e/run_flowgrpo_qwen_image_edit.sh

# Real training run (140+ steps, 8×H200)
bash examples/flowgrpo_trainer/run_qwen_image_edit_lora_pickscore.sh
```

### Data preparation

Convert an image-edit dataset (UniRL format) to Qwen-Image-Edit parquet:

```bash
python3 examples/flowgrpo_trainer/data_process/qwenimageedit_image.py \
    --input_dir ~/data/image_edit \
    --output_dir ~/data/qwen_image_edit
```

### Training

Launch FlowGRPO LoRA training with PickScore reward (local inference mode):

```bash
bash examples/flowgrpo_trainer/run_qwen_image_edit_lora_pickscore.sh
```

### Specific changes

- **Pipeline**: `verl_omni/pipelines/qwen_image_edit_flow_grpo/` — QwenImageEditPlusPipeline with condition-image latent concatenation and SDE window rollout adapter.
- **Reward**: `verl_omni/utils/reward_score/pickscore_reward.py` — local PickScore inference (CLIP-based, async queue consumer).
- **Processor patch**: `verl_omni/utils/hf_processor_patch.py` — auto-patch missing `model_type` in processor `config.json` for Qwen-Image-Edit variants, installed on `verl_omni` import.
- **Data processing**: `examples/flowgrpo_trainer/data_process/qwenimageedit_image.py` — convert UniRL image-edit dataset to parquet.
- **Training script**: `examples/flowgrpo_trainer/run_qwen_image_edit_lora_pickscore.sh` — launch script with validated hyperparameters.
- **E2E test**: `tests/special_e2e/run_flowgrpo_qwen_image_edit.sh` — smoke test with tiny model.
- **Docs**: `docs/algo/flowgrpo.md` — updated image-edit section.

## Next Steps

- Extend FlowGRPO support to FLUX image-edit models.
- Add NFT (Noise-Finetuning Training) support for Qwen-Image-Edit.

## Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
  - ruff and ruff-format passed; full pre-commit (mypy, compileall) deferred to CI.
- [x] Add / Update the documentation.
  - Updated `docs/algo/flowgrpo.md` with image-edit section.
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl-omni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
  - Added `tests/special_e2e/run_flowgrpo_qwen_image_edit.sh` smoke test (tiny model, 1 GPU, 1 step). Validated locally with `tiny-random/qwen-image-edit-plus`.

## AI assistance disclosure

AI assistance (GitHub Copilot / CodeBuddy) was used to draft and prepare parts of this change. The submitter has reviewed every changed line and run the relevant tests.

### CODE REVIEW

File: verl_omni/utils/hf_processor_patch.py
Comment (zhtmike) [inline verl_omni/utils/hf_processor_patch.py:1]: Please drop patch, write a rfc first if you are interested in image editing task.

Comment (NancyFyong) [reply verl_omni/utils/hf_processor_patch.py:1]: Thanks for the feedback. I've written an RFC for the Qwen-Image-Edit FlowGRPO design: #237. Would appreciate your input on the design before resubmitting a patch.
